### ISSUE TITLE
Add basic print styles

### ISSUE THREAD

**Issue by joyously** [Automattic/_s#1207]
Too many authors forget all about print styles. I think having the basic styles of hiding the interactive elements (menu, sidebar, comment form) would not only remind them to handle print styles, but set the standard for the types of things to do for print.

**Comment by philiparthurmoore** [comment]
Big fan of this.

**Comment by samikeijonen** [comment]
I leave basic printing styles off on purpose so that people would not print websites.

**Comment by joyously** [comment]
Haha @samikeijonen !  You know that doesn't keep them from printing, right? Just makes them annoyed that the page doesn't look better for print.
Themes are about presentation, whether it's on the screen or the printer. Think of it like a different side of accessibility.

**Comment by samikeijonen** [comment]
On more serious note it would be nice to know how often printing comes up in wp.com support question. I have had zero which hopefully relates that people don't print that much webpages.

**Comment by joyously** [comment]
That seems tangential. Does the number of support questions for accessibility issues indicate how themes should address accessibility?

**Comment by samikeijonen** [comment]
For me ecological footprint is more important in this case than accessibility. But it's true no one can prevent printing even if it's bad practice. I can also see value in printing in some cases like

- order details
- directions
- recipes

That's if the PDF or better format is not available. 

With that said I'm still voting no to this because print styles would be too generic and bloat the CSS.

**Comment by grappler** [comment]
I don't think if having or not having print styles will make a difference if people print webpages.

By adding printing styles we can reduce the amount of ink that is need and perhaps also the amount of paper that is needed if we reduce redundant information like the widgets in the sidebar.

We can easily create a separate file `print.css` for just some basic print styles. By using the following method we can make sure that the styles are only loaded when printing.

```php
wp_enqueue_style( '_s-print', 'path/print.css', array(), '20170921', 'print' );
```

The print view can be useful for other reasons then print. It may be a format that is easier to read. It can make the PDF's also easier to read.

I think the first step would be to look how the theme looks now when printing then we can discuss what changes would be best.

**Comment by joyously** [comment]
What I'm suggesting is basic styles, like the following. Very short, but useful, and a prompt for the theme author to expand if they want to.
```
@media print {

	/* Hide elements */
	form,
	button,
	input,
	select,
	textarea,
	.navigation,
 	.menu,
	.widget-area,
 	.page-links,
	.edit-link,
	.post-navigation,
	.pagination.navigation,
	.comments-pagination,
	.comment-respond,
	.comment-edit-link,
	.comment-reply-link,
	.comment-metadata .edit-link,
	.pingback .edit-link {
		display: none !important;
	}

  .has-sidebar .content-area {
	float: none;
  	width: 100%;
  }

  html {
	font-size: 85%;  /* this is from a stylesheet with 100% for screen */
  }
}

```

**Comment by philiparthurmoore** [comment]
For something like this screenshots really do help. Do you have visual before and afters?

**Comment by joyously** [comment]
**Here is my print preview with no print styles.** This short post takes two pages to print. (Yes, you can do Shrink to Fit or a percentage in the print options, but there is a lot of clutter.)

![print-without-1](https://user-images.githubusercontent.com/7957680/31150682-342240be-a85a-11e7-9a34-1ec6a0ca6a7d.png)
![print-without-2](https://user-images.githubusercontent.com/7957680/31150683-3426e8a8-a85a-11e7-8d7b-6bc9d208bd67.png)

**Here is my print preview using the above mentioned print styles.** Note that there is less clutter and it takes less than one page to print.

![print-with](https://user-images.githubusercontent.com/7957680/31150681-3417254e-a85a-11e7-8532-71570d5cec98.png)

**Comment by davidakennedy** [comment]
Thanks @joyously for the suggestion on this!

I would be in favor. Seems like a smart thing to do for people, shouldn't take much effort and can be done with performance in mind.

Not sure how often printing comes up in WordPress.com support. We do have [global print styles](https://s0.wp.com/wp-content/mu-plugins/global-print/global-print.css?m=1465851035h&cssminify=no) added to the majority of themes. Keep in mind that we're accounting for a lot more stuff than most theme authors would need to worry about. Looking at HTML5 Boilerplate might be a good idea here, as it has [super simple print styles](https://github.com/h5bp/html5-boilerplate/blob/master/src/css/main.css#L207). Very similar to what @joyously suggests.

**Comment by arnabwahid** [comment]
@joyously @davidakennedy If I add the recommended styles in the theme, in which sub-directory should the print.scss file reside in sass directory? modules? Any suggestions? 

Or enqueuing a separate file as @grappler suggested is the better route?

**Comment by joyously** [comment]
When I add it to a theme, I just put that short block at the end of the normal stylesheet. It's got a media rule, so it only applies to print, and it's short, so it doesn't warrant another http request.
That's what _I_ do, anyway.

**Comment by arnabwahid** [comment]
I agree, should be in the same file and everything else. Thanks @joyously. I am still getting the lay of the land. Suggestions are greatly appreciated! I will send a PR for the print styles soon.

**Comment by davidakennedy** [comment]
@arnabwahid Thanks for asking! I'd probably make a [new section for it here](https://github.com/Automattic/_s/blob/master/sass/style.scss) at the end of the file. And for now, put the new `scss` file in the `site` folder.
